app/controllers/programs/program_managers_controller.rb

```diff
@@ -1,10 +1,68 @@
-class Programs::ProgramManagersController < ProgramManagersController
+class Programs::ProgramManagersController < ApplicationController
   before_action :set_program_manager_program
 
-  private
+  def new
+    @program_manager = ProgramManager.new
+  end
+
+  def create
+    uniqname = params[:program_manager][:uniqname]
+    if ProgramManager.find_by(uniqname: uniqname).present?
+      @manager = ProgramManager.find_by(uniqname: uniqname)
+      if @program_manager_program.program_managers.pluck(:uniqname).include?(uniqname)
+        redirect_to new_program_program_manager_path(@program_manager_program), alert: "#{@manager.display_name} is already a manager"
+        return
+      elsif @program_manager_program.instructor.uniqname == uniqname
+        redirect_to new_program_program_manager_path(@program_manager_program), alert: "#{@manager.display_name} is the instructor"
+        return
+      else
+        @program_manager_program.program_managers << @manager
+        redirect_to @program_manager_program, notice: "The program manager was added"
+        return
+      end
+    else
+      program_manager = ProgramManager.new(program_manager_params)
+      name = LdapLookup.get_simple_name(uniqname)
```

Comment:
If the ldap record is set to 'private' this will return an empty name also.
How else can you √ for a valid uniqname?

Reply:
I don't know now how to check if a uniqname is valid if LdapLookup returns an empty result. But I understand how to create a module for that code. 
I think that should be done in a separate ticket. 
Do you want me to remove LDAP stuff from this pull request?

Reply:
@rsmoke I removed Ldap validation from here to do that in a separate ticket.

Reply:
Since this code was removed, I am resolving this conversation.